odoo/tools/mail.py

```diff
@@ -297,6 +314,13 @@ def is_html_empty(html_content):
     tag_re = re.compile(r'\<\s*\/?(?:p|div|span|br|b|i|font)(?:(?=\s+\w*)[^/>]*|\s*)/?\s*\>')
     return not bool(re.sub(tag_re, '', html_content).strip())
 
+
+def text_to_html_with_urls_and_newlines(text):
```

Comment:
from an API POV, shound't it return a Markup object ?

Reply:
html_keep_url should as well imo (but no need to update it)
but not a big deal anyway

Reply:
Let's rework all that in master, shall we?

Reply:
Maybe I will have to fight, but we should not have 2 methods doing "about the same": we have 'plaintext2html' and we should not have 'plaintext2htmlbutslightlydifferent'. I think the option was a better approach, something like "pretty_html" or stuff like that that triggers additional processing: you can ask for simple html or improve html (with paragraphs management mainly).